### PR TITLE
A Deeper Look Into Git: turning remote section into lesson

### LINKED ISSUE #26042: <JavaScript>: <Splitting Deeper Look at Git>
### Describe your suggestion

Per @wise-king-sullyman's suggestion in [this closed pull request](https://github.com/TheOdinProject/curriculum/pull/25803#issuecomment-1646950912), we should split the the deeper look at Git lesson.

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/javascript-a-deeper-look-at-git

### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### (Optional) Discord Name

Chrissy

### (Optional) Additional Comments

I’d love to work on this with @wise-king-sullyman and @Ba

### PR DESCRIPTION
**Pull request by CBID2**
## Because

Turning the "Working With Remotes" section into its own lesson would make intermediate Git lessons more digestible for learners. 

## This PR

- Turns the "Working With Remotes" section into its own lesson

## Issue

Closes #26042

## Additional Information

This is a work in progress. I have not deleted the "Working Remote" section yet to ensure that the new lesson meets TOP standards. Once I get that approval, then this part will be deleted.

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Comment by Lofty-Brambles** [comment]
Mis-mentioned that, please ignore it :D

**Comment by codyloyd** [comment]
at first glance this looks pretty good to me, but @wise-king-sullyman and @rlmoser99 were much more active in the original discussion that spawned this lesson, so let's get a more in-depth review from them before progressing.

**Comment by CBID2** [comment]
> at first glance this looks pretty good to me, but @wise-king-sullyman and @rlmoser99 were much more active in the original discussion that spawned this lesson, so let's get a more in-depth review from them before progressing.

Sure @codyloyd. This why I placed my PR in draft mode lol 😂

File: git/intermediate_git/working_with_remotes.md
Comment (wise-king-sullyman) [inline git/intermediate_git/working_with_remotes.md:75]: I would keep the merge conflicts reading here.

File: git/intermediate_git/working_with_remotes.md
Comment (wise-king-sullyman) [inline git/intermediate_git/working_with_remotes.md:89]:
```suggestion
* If you're looking for an interactive way to deepen your knowledge of working with Git, check out this game, [Learn Git Branching](https://learngitbranching.js.org/)
```
We don't really know how popular it is, and while I think it's a good tool I don't think many would find it "fun" in the typical sense.

Also the "branches and" seems unneeded since branches are part of Git.

The `the` -> `this` change just makes it read more naturally to me, but if that's grammatically incorrect just ignore this part, I'm no subject matter expert on grammar.

**Review by wise-king-sullyman — COMMENTED** [review]
This is looking pretty good, thanks for doing it!

I just have a few change requests then it should be ready for the final touches.

File: git/intermediate_git/working_with_remotes.md
Comment (CBID2) [reply git/intermediate_git/working_with_remotes.md:75]: Made the changes @wise-king-sullyman.

Comment (wise-king-sullyman) [reply git/intermediate_git/working_with_remotes.md:75]: Do you not want to update the Deeper Look at Git lesson as part of this PR?

Comment (CBID2) [reply git/intermediate_git/working_with_remotes.md:75]: Oh ok I'll gladly do it @wise-king-sullyman

Comment (CBID2) [reply git/intermediate_git/working_with_remotes.md:75]: Made the change @wise-king-sullyman

Comment (wise-king-sullyman) [reply git/intermediate_git/working_with_remotes.md:75]: It looks like you removed the entire bottom half of the Deeper Look at Git lesson? We want to keep the branches are pointers section, the relevant assignments, knowledge checks, and additional resources for that lesson.

Also you should update the lesson overview points.

Comment (CBID2) [reply git/intermediate_git/working_with_remotes.md:75]: Ok @wise-king-sullyman. I'll add those back in. Can you point out which parts I should remove?

Comment (wise-king-sullyman) [reply git/intermediate_git/working_with_remotes.md:75]: All that should be removed from the Deeper Look at Git lesson is the stuff you added to the new remotes lesson.

Comment (CBID2) [reply git/intermediate_git/working_with_remotes.md:75]: Ok

**Comment by ManonLef** [comment]
@CBID2 and @wise-king-sullyman thanks for all your work on this!

FYI:
Before the eventual merge, please also keep in mind [the style guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md). 
I noticed some issues with the list formatting, header capitalisation and newline rules we have in there.

**Comment by CBID2** [comment]
> @CBID2 and @wise-king-sullyman thanks for all your work on this!
> 
> 
> 
> FYI:
> 
> Before the eventual merge, please also keep in mind [the style guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md). 
> 
> I noticed some issues with the list formatting, header capitalisation and newline rules we have in there. 
> 
> 

Thanks @ManonLef! :)

**Comment by CBID2** [comment]
Hey @wise-king-sullyman. I made some of the changes you suggested. I kind of got a little loss with filling out the Assignment and Knowledge check questions, so can you help there?

**Comment by wise-king-sullyman** [comment]
Hey @CBID2, just wanted to let you know that this is on my TODO list but I'm out of town at the moment and quite limited in my availability, I will give your changes a look as soon as I get a chance though!

**Comment by CBID2** [comment]
> Hey @CBID2, just wanted to let you know that this is on my TODO list but I'm out of town at the moment and quite limited in my availability, I will give your changes a look as soon as I get a chance though!

Thanks

**Comment by wise-king-sullyman** [comment]
@CBID2 regarding the assignments I would do what I outlined in [this comment](https://github.com/TheOdinProject/curriculum/issues/26042#issuecomment-1672363745).

For the knowledge checks, I would just split the current ones between the lessons based on the content.

**Comment by CBID2** [comment]
Thanks @wise-king-sullyman! 😊

**Comment by CBID2** [comment]
> 

I think I got this time @wise-king-sullyman! :) 🤞🏾

File: git/intermediate_git/a_deeper_look_at_git.md
Comment (wise-king-sullyman) [inline git/intermediate_git/a_deeper_look_at_git.md:221]: I think this knowledge check should be removed from this lesson.

File: git/intermediate_git/working_with_remotes.md
Comment (wise-king-sullyman) [inline git/intermediate_git/working_with_remotes.md:84]: I feel like this knowledge check goes better with the deeper look at Git lesson.

**Review by wise-king-sullyman — COMMENTED** [review]
A couple small comments but other than those I think the content here is looking pretty good 💪 

As mentioned previously there are some areas where the formatting needs tweaking, but I assume you're planning to address those once you move this out of draft state.

**Comment by CBID2** [comment]
Hi @wise-king-sullyman. I added your suggestions. Regarding formatting, I will need your help because I read the Curriculum Layout sheet and found that the lesson I created conforms to its standards.

**Review by wise-king-sullyman — CHANGES_REQUESTED** [review]
I'd missed it the last time but it looks like when you merged main into your branch you re-added a lot of the content to A Deeper Look at Git that you'd previously removed. Can you go back through, remove the now Working with Remotes content from A Deeper Look at Git, and do another pass on the knowledge checks? It looks like the "dangers of history changing operations" knowledge check is still in both lessons, and some of the relevant ones to A Deeper Look at Git have been removed.

**Comment by CBID2** [comment]
> I'd missed it the last time but it looks like when you merged main into your branch you re-added a lot of the content to A Deeper Look at Git that you'd previously removed. Can you go back through, remove the now Working with Remotes content from A Deeper Look at Git, and do another pass on the knowledge checks? It looks like the "dangers of history changing operations" knowledge check is still in both lessons, and some of the relevant ones to A Deeper Look at Git have been removed. 
> 
> 
> 
> Regarding formatting, I will give that a fine pass this weekend, one thing that stands out to me is that it looks like some spaces were added before the knowledge check heading which I don't think are supposed to be there, but reviewing formatting is something I do a lot better with fresh eyes.

Yeah I got a bit stuck with fixing the knowledge checks @wise-king-sullyman. Can you add your commits as suggestions? It'll help move things quicker

File: git/intermediate_git/a_deeper_look_at_git.md
Comment (wise-king-sullyman) [inline git/intermediate_git/a_deeper_look_at_git.md:230]: The "branches are pointers" knowledge check should be with its content here.

File: git/intermediate_git/working_with_remotes.md
Comment (wise-king-sullyman) [inline git/intermediate_git/working_with_remotes.md:88]: The "amend your last commit" lesson goes into the previous lesson.

**Comment by wise-king-sullyman** [comment]
I can't leave comments for removing the "working with remotes" content that was accidentally added back in as GitHub doesn't let you leave comments in areas it doesn't think are relevant to the current PR.

**Comment by CBID2** [comment]
> I can't leave comments for removing the "working with remotes" content that was accidentally added back in as GitHub doesn't let you leave comments in areas it doesn't think are relevant to the current PR.

Oh ok

**Comment by CBID2** [comment]
> I can't leave comments for removing the "working with remotes" content that was accidentally added back in as GitHub doesn't let you leave comments in areas it doesn't think are relevant to the current PR.

I think I got it this time @wise-king-sullyman. Fingers crossed! 😄  🤞🏾

File: git/intermediate_git/a_deeper_look_at_git.md
Comment (wise-king-sullyman) [inline git/intermediate_git/a_deeper_look_at_git.md:219]: You'll need to add a newline before the first bullet point here otherwise it [won't format properly](https://www.theodinproject.com/lessons/preview?uuid=ba385bbb-95eb-413a-a6e1-d3aaaa9eb7c8).

File: git/intermediate_git/working_with_remotes.md
Comment (wise-king-sullyman) [inline git/intermediate_git/working_with_remotes.md:53]: You'll need a newline here for[ formatting reasons](https://www.theodinproject.com/lessons/preview?uuid=c0126432-aac9-412d-8c1e-79d48d6ad3e0) as well.

**Review by wise-king-sullyman — CHANGES_REQUESTED** [review]
A few additional small items.

Also it looks like the Working with Remotes content is still in A Deeper Look at Git?

File: git/intermediate_git/a_deeper_look_at_git.md
Comment (wise-king-sullyman) [inline git/intermediate_git/a_deeper_look_at_git.md:219]:
```suggestion

* <a class='knowledge-check-link' href='https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell'>Explain what it means for branches to be pointers.</a>
```

File: git/intermediate_git/working_with_remotes.md
Comment (wise-king-sullyman) [inline git/intermediate_git/working_with_remotes.md:53]:
```suggestion

### Dangers and best practices
```

**Comment by CBID2** [comment]
I committed your suggestions @wise-king-sullyman

File: git/intermediate_git/a_deeper_look_at_git.md
Comment (CBID2) [inline git/intermediate_git/a_deeper_look_at_git.md:126]: Hey @wise-king-sullyman. I think this part can be removed.

Comment (CBID2) [inline git/intermediate_git/a_deeper_look_at_git.md:126]: Hey @wise-king-sullyman! :) I think this one can be removed.

File: git/intermediate_git/a_deeper_look_at_git.md
Comment (CBID2) [inline git/intermediate_git/a_deeper_look_at_git.md:128]: This one as well @wise-king-sullyman.

**Review by CBID2 — COMMENTED** [review]
Hey @wise-king-sullyman. I pointed out some parts in the Deeper Look Into Git lesson that can be removed. Check them out and tell me what you think.

**Comment by wise-king-sullyman** [comment]
Yes those sections and the dangers & best practices sections should be removed from A Deeper Look at Git.

**Comment by CBID2** [comment]
> Yes those sections and the dangers & best practices sections should be removed from A Deeper Look at Git.

Done @wise-king-sullyman! :)

**Review by wise-king-sullyman — APPROVED** [review]
Looks great! Now all that's left is to get the corresponding site side change ready and this should be good to merge 🚀 

Is that something you want to do?

**Comment by CBID2** [comment]
> Looks great! Now all that's left is to get the corresponding site side change ready and this should be good to merge 🚀 
> 
> 
> 
> Is that something you want to do?

Sure @wise-king-sullyman! :)

**Comment by wise-king-sullyman** [comment]
@CBID2 have you done any work on our site repo before?

**Comment by CBID2** [comment]
> @CBID2 have you done any work on our site repo before?

No I haven't @wise-king-sullyman. I'm eager to learn though 😊

**Comment by wise-king-sullyman** [comment]
Just as a heads up, the site is a _much_ more complex repo to make changes to.

Broadly I would point you to[ the wiki](https://github.com/TheOdinProject/theodinproject/wiki), and specifically the pages about getting the app to run locally so that you know the changes you make will work before you put a PR up, and the page about adding a lesson as that's what you'll need to do to get your new lesson displayed.

**Comment by CBID2** [comment]
> Just as a heads up, the site is a _much_ more complex repo to make changes to. If you don't have any prior exposure to Rails apps it can seem like a _lot_. I'm often nervous to jump into the site myself and I've been doing software development professionally for a few years now 😆
> 
> Broadly I would point you to[ the wiki](https://github.com/TheOdinProject/theodinproject/wiki), and specifically the pages about getting the app installed and able to run locally so that you know the changes you make will work before you put a PR up, and the page about adding a lesson as that's what you'll need to do to get your new lesson displayed.

Hi @wise-king-sullyman! :) I'm on the [[Set Up DotEnv](https://github.com/TheOdinProject/theodinproject/wiki/Running-The-Odin-Project-Locally#set-up-dotenv), and when I try to rename `$ cp env.sample .env` as `$ cp env.env .env`, I get this response:
`cp: env.env: No such file or directory`
How should I rename the env file then?

**Comment by wise-king-sullyman** [comment]
`$ cp env.sample .env` is the command to run. It copies the `env.sample` file and names that copy `.env`.

**Comment by CBID2** [comment]
> `$ cp env.sample .env` is the command to run. It copies the `env.sample` file and names that copy `.env`.

Ooh ok @wise-king-sullyman.

**Comment by CBID2** [comment]
> `$ cp env.sample .env` is the command to run. It copies the `env.sample` file and names that copy `.env`.

Now my issue is that when I do `find /Applications -name pg_config`, the path does not print in my terminal. Do you recommend that create a pg_config file @wise-king-sullyman?

**Comment by CBID2** [comment]
> > `$ cp env.sample .env` is the command to run. It copies the `env.sample` file and names that copy `.env`.
> 
> Now my issue is that when I do `find /Applications -name pg_config`, the path does not print in my terminal. Do you recommend that create a pg_config file @wise-king-sullyman? 

@ManonLef, can you help?

**Comment by ManonLef** [comment]
@CBID2 I’d rather leave this to the maintainers involved and who have more experience with this 😊

**Comment by CBID2** [comment]
> @CBID2 I’d rather leave this to the maintainers involved and who have more experience with this 😊

Ok thanks @ManonLef

**Comment by CBID2** [comment]
> `$ cp env.sample .env` is the command to run. It copies the `env.sample` file and names that copy `.env`.

Hi @wise-king-sullyman! :) I managed to set up TOP locally. Now, do you recommend that I follow the [Adding a New Lesson section](https://github.com/TheOdinProject/theodinproject/wiki/Adding-a-Lesson)?

**Comment by wise-king-sullyman** [comment]
@CBID2 correct

**Comment by CBID2** [comment]
> @CBID2 correct

Great thanks! :)

**Comment by CBID2** [comment]
Hi @wise-king-sullyman! :) I created the PR in the website repo. It's (https://github.com/TheOdinProject/theodinproject/pull/4202). I added some notes in the Additional Information section, so please check them out.

**Pull request merged**
